pkg/controller/atlasdeployment/region_configuration.go

```diff
@@ -39,9 +39,20 @@ func syncRegionConfiguration(deploymentSpec *mdbv1.AdvancedDeploymentSpec, atlas
 		// When compute auto-scaling is enabled, unset instance size to avoid override production workload
 		if isComputeAutoScalingEnabled(regionSpec.AutoScaling) {
 			if !regionsHasChanged {
-				regionSpec.ElectableSpecs.InstanceSize = ""
-				regionSpec.ReadOnlySpecs.InstanceSize = ""
-				regionSpec.AnalyticsSpecs.InstanceSize = ""
+				for _, arc := range atlasCluster.ReplicationSpecs[0].RegionConfigs {
```

Comment:
nit: this will panic if `len(atlasCluster.ReplicationSpecs) == 0`. have we checked that this is slice is >0?

Reply:
Such config will not pass validation. But let me double check.

Reply:
Yep, it is rejected at apply time:
`The AtlasDeployment ... is invalid: spec.deploymentSpec.providerSettings: Required value`